[0.009s] Invoking command in '/home/simpletime/project01_book/build/base_interfaces': CMAKE_PREFIX_PATH=${CMAKE_PREFIX_PATH}:/opt/ros/humble /usr/bin/cmake --build /home/simpletime/project01_book/build/base_interfaces -- -j4 -l4
[0.027s] -- Found ament_cmake: 1.3.10 (/opt/ros/humble/share/ament_cmake/cmake)
[0.118s] -- Found rosidl_default_generators: 1.2.0 (/opt/ros/humble/share/rosidl_default_generators/cmake)
[0.119s] -- Using all available rosidl_typesupport_c: rosidl_typesupport_fastrtps_c;rosidl_typesupport_introspection_c
[0.120s] -- Found rosidl_adapter: 3.1.5 (/opt/ros/humble/share/rosidl_adapter/cmake)
[0.137s] -- Using all available rosidl_typesupport_cpp: rosidl_typesupport_fastrtps_cpp;rosidl_typesupport_introspection_cpp
[0.256s] [31mCMake Error at /opt/ros/humble/share/rosidl_adapter/cmake/rosidl_adapt_interfaces.cmake:59 (message):
[0.256s]   execute_process(/usr/bin/python3 -m rosidl_adapter --package-name
[0.256s]   base_interfaces --arguments-file
[0.256s]   /home/simpletime/project01_book/build/base_interfaces/rosidl_adapter__arguments__base_interfaces.json
[0.256s]   --output-dir
[0.256s]   /home/simpletime/project01_book/build/base_interfaces/rosidl_adapter/base_interfaces
[0.256s]   --output-file
[0.256s]   /home/simpletime/project01_book/build/base_interfaces/rosidl_adapter/base_interfaces.idls)
[0.256s]   returned error code 1:
[0.256s] 
[0.256s]   Traceback (most recent call last):
[0.256s] 
[0.256s]     File "/usr/lib/python3.10/runpy.py", line 196, in _run_module_as_main
[0.256s]       return _run_code(code, main_globals, None,
[0.256s]     File "/usr/lib/python3.10/runpy.py", line 86, in _run_code
[0.256s]       exec(code, run_globals)
[0.256s]     File "/opt/ros/humble/local/lib/python3.10/dist-packages/rosidl_adapter/__main__.py", line 19, in <module>
[0.256s]       sys.exit(main())
[0.256s]     File "/opt/ros/humble/local/lib/python3.10/dist-packages/rosidl_adapter/main.py", line 53, in main
[0.257s]       abs_idl_file = convert_to_idl(
[0.257s]     File "/opt/ros/humble/local/lib/python3.10/dist-packages/rosidl_adapter/__init__.py", line 19, in convert_to_idl
[0.257s]       return convert_msg_to_idl(
[0.257s]     File "/opt/ros/humble/local/lib/python3.10/dist-packages/rosidl_adapter/msg/__init__.py", line 28, in convert_msg_to_idl
[0.257s]       msg = parse_message_string(package_name, input_file.stem, content)
[0.257s]     File "/opt/ros/humble/local/lib/python3.10/dist-packages/rosidl_adapter/parser.py", line 544, in parse_message_string
[0.257s]       msg = MessageSpecification(pkg_name, msg_name, fields, constants)
[0.257s]     File "/opt/ros/humble/local/lib/python3.10/dist-packages/rosidl_adapter/parser.py", line 388, in __init__
[0.257s]       self.base_type = BaseType(
[0.257s]     File "/opt/ros/humble/local/lib/python3.10/dist-packages/rosidl_adapter/parser.py", line 205, in __init__
[0.257s]       raise InvalidResourceName(
[0.257s] 
[0.257s]   rosidl_adapter.parser.InvalidResourceName: 'Book_need' is an invalid
[0.257s]   message name.  It should have the pattern '^[A-Z][A-Za-z0-9]*$'
[0.257s] 
[0.257s] Call Stack (most recent call first):
[0.257s]   /opt/ros/humble/share/rosidl_cmake/cmake/rosidl_generate_interfaces.cmake:130 (rosidl_adapt_interfaces)
[0.257s]   CMakeLists.txt:16 (rosidl_generate_interfaces)
[0.257s] 
[0.257s] [0m
[0.257s] -- Configuring incomplete, errors occurred!
[0.257s] See also "/home/simpletime/project01_book/build/base_interfaces/CMakeFiles/CMakeOutput.log".
[0.261s] gmake: *** [Makefile:226ï¼šcmake_check_build_system] é”™è¯¯ 1
[0.264s] Invoked command in '/home/simpletime/project01_book/build/base_interfaces' returned '2': CMAKE_PREFIX_PATH=${CMAKE_PREFIX_PATH}:/opt/ros/humble /usr/bin/cmake --build /home/simpletime/project01_book/build/base_interfaces -- -j4 -l4
